ocs_ci/ocs/resources/storageconsumer.py

```diff
@@ -0,0 +1,120 @@
+"""
+A module for all StorageConsumer functionalities and abstractions.
+"""
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import ocp
+
+log = logging.getLogger(__name__)
+
+
+class StorageConsumer:
+    """
+    Base StorageConsumer class
+    """
+
+    def __init__(self, consumer_name, consumer_context=None):
+        """
+        Args:
+            consumer_name (string): name of the StorageConsumer resource
+            consumer_context (int): index of cluster context. This is needed for
+                consumer operations executed on consumer
+                (e.g. manipulation of heartbeat cronjob)
+        """
+        self.consumer_context = consumer_context
+        self.name = consumer_name
+        self.ocp = ocp.OCP(
+            resource_name=self.name,
+            kind=constants.STORAGECONSUMER,
+            namespace=config.cluster_ctx.ENV_DATA["cluster_namespace"],
+        )
+        if self.consumer_context:
+            self.heartbeat_cronjob = self.get_heartbeat_cronjob()
+            self.original_context = config.cluster_ctx
```

Comment:
maybe better to name self.original_context as provider_context, that should go well with existing self.consumer_context